src/super_gradients/training/datasets/dali_datasets/dali_pipelines.py

```diff
@@ -0,0 +1,77 @@
+try:
+    from nvidia.dali.pipeline import pipeline_def
+    import nvidia.dali.types as types
+    import nvidia.dali.fn as fn
+except ImportError:
+    raise ImportError("Please install DALI from https://www.github.com/NVIDIA/DALI.")
+
+
+@pipeline_def
+def imagenet_dali_pipeline(data_dir: str, crop: int, resize_size: int, shard_id: int, num_shards: int, dali_cpu=False, is_training=True):
```

Comment:
@shaydeci  - I think that this design pattern by nvidia could be easily modified to a "DaliDataSetInterface"... 

I would expect that Dali integration could be implemented generically so it can wrap every dataset interface we already have in deci, and this will be much cleaner, leaner and less error prone.

Also - It will be more future-proof, and will be much easier to maintain.